src/commonMethods/webKitPerformanceCommonFunctions.js

```diff
@@ -0,0 +1,61 @@
+let samples = []
+
+/**
+ * This function resets the Samples array
+ */
+export const initializeSamplesArray = function() {
+  samples = []
+}
+
+/**
+ * This function fetches the FPS value
+ * @returns {Samples Array}
+ */
+export const fetchFPS = function() {
+  this.$thunder.api.WebKitBrowser.fps()
+    .then(result => {
+      if (isNaN(result) === false) {
+        samples.push(result)
+        return true
+      } else {
+        this.$log('Returned FPS value from Framework is not a number')
+        throw new Error('Returned FPS value from Framework is not a number')
```

Comment:
not sure if `fetchFPS` should throw and Error or just ignore a possible `NaN`-entry .. Thoughts @wouterlucas?

Reply:
We might be a little defensive here, as the test is not to validate if the FPS counter works but to validate if the page being run meets a minimum amount of FPS.

I'd not throw an error, just ignore and continue.

Reply:
Updated the code as per the comment by removing throw error